server/wfprev-api/src/main/java/ca/bc/gov/nrs/wfprev/services/ActivityService.java

```diff
@@ -0,0 +1,233 @@
+    }
+
+    private ActivityModel saveActivity(ActivityEntity entity) {
+        try {
+            ActivityEntity savedEntity = activityRepository.saveAndFlush(entity);
+            return activityResourceAssembler.toModel(savedEntity);
+        } catch (DataIntegrityViolationException | ConstraintViolationException e) {
+            log.error("Data integrity or constraint violation: {}", e.getMessage(), e);
+            throw e;
+        } catch (EntityNotFoundException e) {
+            log.error("Invalid reference data: {}", e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    public ActivityModel getActivity(String projectGuid, String fiscalGuid, String activityGuid) {
+        // Get the activity
+        ActivityEntity activity = activityRepository.findById(UUID.fromString(activityGuid))
+                .orElseThrow(() -> new EntityNotFoundException(ACTIVITY_NOT_FOUND + activityGuid));
+
+        // Verify activity belongs to the specified project fiscal
+        if (!activity.getProjectPlanFiscalGuid().toString().equals(fiscalGuid)) {
+            throw new EntityNotFoundException(ACTIVITY + activityGuid + DOES_NOT_BELONG_FISCAL + fiscalGuid);
+        }
+
+        // Verify project fiscal belongs to project
+        ProjectFiscalEntity projectFiscal = projectFiscalRepository.findById(UUID.fromString(fiscalGuid))
+                .orElseThrow(() -> new EntityNotFoundException(FISCAL_NOT_FOUND + fiscalGuid));
+        if (!projectFiscal.getProject().getProjectGuid().toString().equals(projectGuid)) {
+            throw new EntityNotFoundException(PROJECT_FISCAL + fiscalGuid + DOES_NOT_BELONG_PROJECT + projectGuid);
+        }
+
+        return activityResourceAssembler.toModel(activity);
+    }
+
+    public void deleteActivity(String projectGuid, String fiscalGuid, String activityGuid) {
+        // Get the activity
+        ActivityEntity activity = activityRepository.findById(UUID.fromString(activityGuid))
+                .orElseThrow(() -> new EntityNotFoundException(ACTIVITY_NOT_FOUND + activityGuid));
+
+        // Verify activity belongs to the specified project fiscal
+        if (!activity.getProjectPlanFiscalGuid().toString().equals(fiscalGuid)) {
+            throw new EntityNotFoundException(ACTIVITY + activityGuid + DOES_NOT_BELONG_FISCAL + fiscalGuid);
+        }
+
+        // Verify project fiscal belongs to project
+        ProjectFiscalEntity projectFiscal = projectFiscalRepository.findById(UUID.fromString(fiscalGuid))
+                .orElseThrow(() -> new EntityNotFoundException(FISCAL_NOT_FOUND + fiscalGuid));
+        if (!projectFiscal.getProject().getProjectGuid().toString().equals(projectGuid)) {
+            throw new EntityNotFoundException(PROJECT_FISCAL + fiscalGuid + DOES_NOT_BELONG_PROJECT + projectGuid);
+        }
+
+        activityRepository.deleteById(UUID.fromString(activityGuid));
+    }
+
+    public void assignAssociatedEntities(ActivityModel resource, ActivityEntity entity) {
```

Comment:
Binding these codes to objects from resource to entity should be handled by hibernate already. Thats what the `@ManyToOne(fetch = FetchType.EAGER, optional = true)` bits are doing. Why are we manually applying them here from the same value? What am I missing?

Reply:
I think assigning them manually is needed as when I break at line 197, the code models do not have each field populated, just their relevant code string
![image](https://github.com/user-attachments/assets/f55a2cea-4c93-46a8-95e4-2052e715d9e1)
When these codes do not get manually assigned the insert gives the error org.hibernate.TransientPropertyValueException: object references an unsaved transient instance - save the transient instance before flushing : ca.bc.gov.nrs.wfprev.data.entities.ActivityEntity.activityStatusCode -> ca.bc.gov.nrs.wfprev.data.entities.ActivityStatusCodeEntity

Reply:
I think the manual assignment is needed because the code model attributes are not fully assigned at this point, aside from the code string itself
![image](https://github.com/user-attachments/assets/1ea1e2a7-6fda-448d-b2f5-0ffef29cf870)
which results in `org.hibernate.TransientPropertyValueException: object references an unsaved transient instance - save the transient instance before flushing : ca.bc.gov.nrs.wfprev.data.entities.ActivityEntity.activityStatusCode -> ca.bc.gov.nrs.wfprev.data.entities.ActivityStatusCodeEntity`
This is also being done in the ProjectService

Reply:
Ah. This will be an issue with the relationship fetch/cascade rules. We can look into that later